src/nupic/research/temporal_memory.py

```diff
@@ -206,6 +212,20 @@ def compute(self, activeColumns, learn=True):
                                                self.predictedSegmentDecrement,
                                                prevActiveCells)
 
+
+  def activateDendrites(self, learn=True):
```

Comment:
I like you've split this into a separate method, but now it's called at the beginning of `compute`, whereas this computation was previously at the end (after TM cells states have been changed for this time step). Why?